debian/cloud-init-base.postinst

```diff
@@ -269,43 +264,52 @@ move_socket_fifo_to_share_dir() {
     fi
 }
 
-# Migrate a debconf value from cloud-init/* to cloud-init-base/*
+
+# Process cloud-init-base debconf values and cloud-init values, emptying or
+# unregistering after the value has been processed.
+#
+# Due to package rename from cloud-init to cloud-init-base failing
+# to unregister old cloud-init/* debconf values across package upgrade
+# we can no longer unregister cloud-init/* values. Also MAAS servers <= 3.3
+# only set cloud-init/* debconf seed values. We can drop the db_set "$old"
+# after MAAS 3.3 is out of support.
 #
-# Arguments:
-#   Name of the debconf question
-migrate_debconf_to_cloud_init_base() {
+# MAAS 3.6 no longer provides debconf settings, but instead uses curtin to
+# write /etc/cloud/cloud.cfg,d/95-cloudconfig-* files directly. When 3.3 is out
+# of support, cloud-init-base can drop support for handling any debconf values.
+process_debconf_setting() {
     local old="cloud-init/$1" new="cloud-init-base/$1"
-    if db_get "$new"; then
-       # cloud-init-base/* win against cloud-init/* values
-       return 0
+    db_get "$old" && old_val="$RET" || old_val=
+    db_get "$new" && new_val="$RET" || new_val=
```

Comment:
Is the last assignment valid syntax? If so, I'm unfamiliar with it. Can we use an alternative, please?

Reply:
Sure, I've dropped the `||`  clause and initialized the old_val and new_val to empty strings on the line above. We will only set them to `$RET` if db_get finds $old or $new settings.